### PR TITLE
Non UI Projects as .netstandard2.0 libraries

### PR DESCRIPTION
**Pull request by UliEgger**
Solves Issue 1479

### Checklist

- [x ] I have included examples or tests: No example needed. Solution must compile. Unit tests and DemoMap can be used
- [ x] I have updated the change log : Yes
- [x ] I am listed in the CONTRIBUTORS file : Yes
- [ x] I have cleaned up the commit history (use rebase and squash) : Yes

### Changes proposed in this pull request:

Projects that are not UI Projects are converted to .netstandard 2.0 libraries
These can be flexibly used in

- Other netstandard libraries
- NET 6 libraries and applications
- NET Framwork 4.8 libraries and applications

Note that some unit tests in OpenFileThrowsRightException fail because of Assertion in AttributeTable.Open
But this happened also before this change (should be another issue)

### CODE REVIEW

File: Source/Core/DotSpatial.Data/DotSpatial.Data.csproj [hunk 1]
```diff
@@ -1,15 +1,18 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0-windows</TargetFramework>
-    <Description>GIS Data objects for DotSpatial.</Description>
+      <LangVersion>9.0</LangVersion>
```
Comment (jany-tenaj) [inline Source/Core/DotSpatial.Data/DotSpatial.Data.csproj:4]: why not 10.0?

File: Source/Core/DotSpatial.Projections/DotSpatial.Projections.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		<TargetFrameworks>net6.0;netstandard2.0</TargetFrameworks>
+		<TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (jany-tenaj) [inline Source/Core/DotSpatial.Projections/DotSpatial.Projections.csproj:4]: why did you remove net6.0?

File: Source/Core/DotSpatial.Data/DotSpatial.Data.csproj (hunk 1 above)
Comment (UliEgger) [reply Source/Core/DotSpatial.Data/DotSpatial.Data.csproj:4]: Language Version 9 was required, but it could be also 10.
The problem was that for .netstandard, the Language Version was 7.3 by default. And then, there was code that did not compile.

File: Source/Core/DotSpatial.Projections/DotSpatial.Projections.csproj (hunk 2 above)
Comment (UliEgger) [reply Source/Core/DotSpatial.Projections/DotSpatial.Projections.csproj:4]: The reason is that multitargeting for this project is not required. NetStandard library will run with both .NET 6 Application and other netstandard libraries.

**Comment by UliEgger** [comment]
I added a commit that switches also GdalExtension to netstandard

**Comment by jany-tenaj** [comment]
@UliEgger build failed

**Comment by UliEgger** [comment]
I reverted my last change again. It was not correct since the project depends on .net 6 project DotSpatial.Controls

**Comment by UliEgger** [comment]
We would need new adaptions because of Commit  5c7dbe931466411e5ca7e4a3d62da20bddc0e868

System.Range is not supported in a .netstandard project without declaring System.Range and System.Index

There would be a way as proposed here

https://www.meziantou.net/how-to-use-csharp-8-indices-and-ranges-in-dotnet-standard-2-0-and-dotn.htm

But since it is only one line of code affected, it seems to be easier to use the old syntax for this line.

**Comment by jany-tenaj** [comment]
Seems we should merge this before we do anything else so we don't create more things you'll have to revert. But I sure don't like being forced to stick to old syntax.

**Comment by UliEgger** [comment]
> Seems we should merge this before we do anything else so we don't create more things you'll have to revert. But I sure don't like being forced to stick to old syntax.

I will ask my team about the best way to allow this new language features within netstandard projects.
